devito/archinfo.py

```diff
@@ -107,15 +113,31 @@ def get_cpu_brand():
         # Fallback 1: it should now be fine to use psutil
         physical = psutil.cpu_count(logical=False)
         if not physical:
-            # Fallback 2: we might end up here on more exotic platforms such a Power8
-            # Hopefully we can rely on `lscpu`
             try:
-                physical = lscpu()['Core(s) per socket'] * lscpu()['Socket(s)']
+                lscpu_mapper = lscpu()
+                # Fallback 2: we might end up here
+                # on more exotic platforms such a Power8 or due to
+                # erroneous autodetection
+                physical = (lscpu_mapper['Core(s) per socket']
+                            * lscpu_mapper['Socket(s)'])
             except KeyError:
                 warning("Physical core count autodetection failed")
                 physical = 1
-    cpu_info['physical'] = physical
 
+    try:
+        if cpu_info['brand'] == 'aarch64':
+            # In some ARM processors psutils and lscpu fail to detect cores correctly
+            physical = psutil.cpu_count(logical=False)
+            lscpu_mapper = lscpu()
+            if physical != (lscpu_mapper['Core(s) per socket']
+                            * lscpu_mapper['Socket(s)']):
+                warning("Platform core count detection is not reliable. Using lscpu()")
+                physical = (lscpu_mapper['Core(s) per socket']
+                            * lscpu_mapper['Socket(s)'])
+    except:
```

Comment:
out of curiosity, we don't know when we could end up here right?

Reply:
To the best of my knowledge, no.